cores/arduino/UART.cpp

```diff
@@ -114,7 +115,7 @@ void HardwareSerial::_tx_udr_empty_irq(void)
 
 // Public Methods //////////////////////////////////////////////////////////////
 
-void HardwareSerial::begin(unsigned long baud, byte config)
+void UartClass::begin(unsigned long baud, uint16_t config)
```

Comment:
The `config` is later assigned to a `*_ucsrc` which is declared as a `uint8_t` in UART.h. Was this change intentional?

Reply:
This is due to https://github.com/arduino/ArduinoCore-API/blob/master/api/HardwareSerial.h#L91 , so the signature must be `uint16_t` .
Said that, I'll push a commit to fix this by casting the assignment
`*_ucsrc = (uint8_t)config;` (since we are assured that config fits an uint8_t)